R/event_manager.R

```diff
@@ -90,6 +125,42 @@ EventManager <- R6Class(
     # notifications, others do not. (Not really sure why.)
     event_enable_domains = NULL,
 
+    event_domain_manually_enabled = list(),
+
+    maybe_event_domain_enable = function(domain) {
+      # If we're doing auto events
+      if (!private$session$get_auto_events()) return()
+      # and this domain requires events (not all domains require or have an .enable method)
+      if (!isTRUE(private$event_enable_domains[[domain]])) return()
+      # and events were not already manually enabled
+      if (isTRUE(private$event_domain_manually_enabled[[domain]])) return()
+      # and we're going from 0 to 1 callbacks in this domain
+      # or to 1+ the number of callbacks when `$disable()` was last called
+      threshold <- max(c(1, private$event_callback_count_threshold[[domain]]))
+      if (private$event_callback_counts[[domain]] != threshold) return()
```

Comment:
This part accounts for a user calling, e.g. `session$Fetch$disable()`, before we've received all of the expected events. In other words, before we've been able to decrement the callback counter back down to `0`. When `$disable()` is called, we store the current count callback + 1, which becomes the new threshold for re-enabling auto events.